src/proxy_handler.cpp

```diff
@@ -0,0 +1,100 @@
+#include <string>
+#include <istream>
+#include <ostream>
+#include <boost/asio.hpp>
+#include "proxy_handler.h"
+
+using boost::asio::ip::tcp;
+RequestHandler::Status ProxyHandler::Init(const std::string& uri_prefix,
+         const NginxConfig& config) {
+	uri_prefix_ = uri_prefix;
+	RequestHandler::ParseConfig(config);
+	host = ProxyHandler::getHost();
+	portno = ProxyHandler::getPort();
+
+	return OK;
+}
+
+RequestHandler::Status ProxyHandler::HandleRequest(const Request& request,
+		Response* response) {
+    //This function is inspired by www.boost.org/doc/libs/1_49_0/doc/html/boost_asio/example/http/client/sync_client.cpp
+    std::string uri;
+    if (request.uri() == uri_prefix_)
+    uri = "/" ;
+    else
+    uri = request.uri();
+
+    if (uri != uri_prefix_ 
+        && uri.size() > uri_prefix_.size() 
+        && uri.substr(0, uri_prefix_.size()) == uri_prefix_) {
+      uri=  uri.substr(uri_prefix_.size());
+    } 
+  
+	boost::asio::io_service io_service;
+    tcp::resolver resolver(io_service);
+    tcp::resolver::query query(host, portno);
+    tcp::resolver::iterator endpoint_iterator = resolver.resolve(query);
+    tcp::socket socket(io_service);
+    boost::asio::connect(socket, endpoint_iterator);
+    
+    std::string req = "GET " + uri + " HTTP/1.1\r\n";
```

Comment:
Please just add a few comments.
// Build and send request
// Read and process response
etc.